learn-ocaml.el

```diff
@@ -774,52 +819,122 @@ Note: this function will be used by `learn-ocaml-on-load-aux'."
           nil
           callback))))))
 
-(defun learn-ocaml-on-load-aux (token server callback)
+(cl-defun learn-ocaml-init-cmd (&key token server nickname secret callback)
+  "Run \"learn-ocaml-client init\" with options."
+  (learn-ocaml-print-time-stamp)
+  (learn-ocaml-make-process-wrapper
+   :name "init"
+   :command (learn-ocaml-command-constructor
+             :token token
+             :server server
+             :param1 nickname
+             :param2 secret
+             :command "init")
+   :stderr (learn-ocaml-log-buffer)
+   :sentinel (apply-partially
+              #'learn-ocaml-error-handler
+              nil
+              callback)))
+
+
+
+(defun learn-ocaml-connection (server callback)
+  "Connect the user with a (login,passwd) or a token and continue with the callback"
+             (cl-case (x-popup-dialog
+                       t `("Welcome to Learn OCaml mode for Emacs.\nWhat do you to do?\n"
+                           ("Sign in" . 1)
+                           ("Sign up" . 2)
+                           ("Connect with an old token" . 3)))
+               (1 (let* ((login_password (learn-ocaml-sign-in))
+                         (login (nth 0 login_password))
+                         (password (nth 1 login_password))
+                         (message
+                          (learn-ocaml-client-sign-in-cmd server login password)))))
+               (2 (let* ((infos (learn-ocaml-sign-up))
+                         (login (nth 0 infos))
+                         (password (nth 1 infos))
+                         (nickname (nth 2 infos))
+                         (secret (nth 3 infos))
+                         (message (learn-ocaml-client-sign-up-cmd
+                                   server login password nickname (escape-secret secret))))
+                    (message-box message)
+                    (learn-ocaml-connection server callback)))
+               (3 (let ((token (read-string "Enter token: ")))
+                    (learn-ocaml-use-metadata-cmd
+                     token
+                     nil
+                     (lambda (_)
+                       (message-box "Token saved."))))))
+             (funcall callback))
```

Comment:
This seems correct, but maybe incomplete w.r.t. error handling (typically, if use_passwd=true and the account has been "upgraded" to login/passwd).

But no need to add unit tests for this now…
one should just do some manual check / fix / acceptation tests for now I guess.